app/controllers/users/invitations_controller.rb

```diff
@@ -1,29 +1,44 @@
 class Users::InvitationsController < Devise::InvitationsController
-  before_action :authorise_manage_team, only: %i(create new)
   before_action :delete_user_record, if: :user_should_be_cleared?, only: :create
-  before_action :add_organisation_to_params, :validate_invited_user, only: :create
-
+  before_action :return_user_to_invite_page, if: :user_is_invalid?, only: :create
+  before_action :add_organisation_to_params, only: :create
 
 private
 
+  def authenticate_inviter!
+    # https://github.com/scambra/devise_invitable#controller-filter
+    redirect_to(root_path) unless current_user&.can_manage_team?
+  end
+
+  def add_organisation_to_params
+    params[:user][:organisation_id] = current_user.organisation_id
+  end
+
   def delete_user_record
-    User.find_by(email: invite_params[:email]).destroy!
+    invited_user.destroy!
   end
 
-  def resending_invite?
-    !!params[:resend]
+  def return_user_to_invite_page
+    respond_with_navigational(resource) { render :new }
   end
 
-  def after_invite_path_for(_resource)
-    team_members_path
+  def user_is_invalid?
+    # This is an indirect solution to preventing a user being re-invited when they belong
+    # to another organisation.
+    self.resource = resource_class.new(invite_params)
+    resource.invalid?
```

Comment:
I preferred the old syntax where it was explicitly testing a user. The `resource` abstraction is a Devise one because Devise allows you to name your user model whatever you like. We use `User` so I think we should be explicit. Eg:
```
  @user = User.new(invite_params)
  !@user.validate
```
or `@user.invalid?` as you've used.